src/adapter/src/coord/ddl.rs

```diff
@@ -380,11 +379,12 @@ impl Coordinator {
                 tracing::error!("Instructed to drop a non-index index");
             }
         }
+        let mut compute = self.controller.active_compute();
         for (cluster_id, ids) in by_cluster {
-            self.controller
-                .active_compute()
-                .drop_collections(cluster_id, ids)
-                .unwrap();
+            // A cluster could have been dropped, so verify it exists.
+            if compute.instance_exists(cluster_id) {
+                compute.drop_collections(cluster_id, ids).unwrap();
+            }
```

Comment:
I couldn't think of a reason why indexes should be different than materialized views or subscribes here, so I added the check for the cluster's existence. Let me know if you think the correct action would be to remove this check from all the compute `drop_` methods instead!